### ISSUE TITLE
regression: Blank black screen when booting into unstable ISO (minimal/plasma/gnome) after flashing with Rufus

### ISSUE THREAD

**Issue by evalgrind** [NixOS/nixpkgs#245915]
### Describe the bug
I only get a blank black screen when booting in the latest NixOS unstable ISO from releases.nixos.org.
I've tried the plasma, gnome and minimal version and all three seem to have the same issue. I'm running this
on Dell Latitude 7300 (Touchscreen, Secure Boot, Fastboot, TPM disabled) with intel iGPU. (no nVidia).

### Steps To Reproduce
1. Flash the latest unstable ISO from releases.nixos.org
2. When in Rufus, use GPT with FAT32 and flash the image to a USB drive.
3. Once booted, now you are presented with a black blank screen with no keyboard support.

### Expected behavior
Booting into the ISO without any issues.

### Additional context
I've looked into other solutions however they're mostly related to nVidia or BIOS settings issues which I do not have.
This problem becomes similar to [24501](https://github.com/NixOS/nixpkgs/issues/245101) if I use Ventoy instead of Rufus.

### Notify maintainers
@K900 (updater), @samueldr (maintainer)

### Metadata
Not applicable.

**Comment by samueldr** [comment]
The only supported method of using the ISO is to copy byte-for-byte on the media.

There are loads of reasons it is hard to support other schemes, mainly because they have to work around distro idiosyncrasies to actually achieve what they do.

Can you verify that it works when using e.g. etcher or win32imager (assuming you can't use `dd`)?

**Comment by evalgrind** [comment]
Rufus with DD Image mode, win32imager and etcher all result with a blank black screen. When using the dd method, it creates two boot options, and I've tried booting into both but both show a black screen, nothing else.

**Comment by K900** [comment]
To clarify, you're not getting a GRUB menu at all, just a black screen? Or are you getting GRUB, choosing to boot into the installer, and _then_ getting a black screen?

**Comment by evalgrind** [comment]
I only get a black screen, no GRUB menu or anything after booting from the USB.

**Comment by K900** [comment]
Can you try rebuilding GRUB with https://file.savannah.gnu.org/file/smartmem.patch?file_id=54987 ?

**Comment by evalgrind** [comment]
I'm using a release ISO, didn't built one so I dont know how to rebuild GRUB.

**Comment by K900** [comment]
Try replacing the GRUB files on the USB with these: https://static.0upti.me/grub.txz

**Comment by evalgrind** [comment]
What specifically to replace? It seems to be some sort of root directory from what you've sent.

**Comment by evalgrind** [comment]
Also, upon deeper inspection the booting process seems to get hung up on loading `EFI/boot/bootx64.efi`. Thought this might narrow _something_.

**Comment by samueldr** [comment]
That would be GRUB in the ISO.

**Comment by evalgrind** [comment]
> That would be GRUB in the ISO.

So `boot/grub` or `EFI/boot`?

**Comment by evalgrind** [comment]
Nope, still the same result even after replacing the GRUB files.

**Comment by K900** [comment]
This issue seems related: https://savannah.gnu.org/bugs/?64294

Maybe leave a comment there.

**Comment by evalgrind** [comment]
Alright, i added the comment. Is there a possible workaround or this is supposed to be fixed by the GRUB devs?

**Comment by K900** [comment]
We're looking at getting rid of GRUB as a bootloader in the images, but this is a work in progress.

**Comment by evalgrind** [comment]
Any pull request/issue I can use to check the progress, I'm guessing it's replaced by systemd-boot?

**Comment by RaitoBezarius** [comment]
I built a systemd-boot ISO image here: https://github.com/NixOS/nixpkgs/pull/246441

To test, just check out this branch and run `nix-build nixos/release.nix -A iso_minimal.x86_64-linux` for example, you will get a minimal ISO for x86_64-linux with a systemd-boot for UEFI.

This is experimental, and I am interested to understand if this fixes the issue for you, I can also provide you with this pre-built ISO: https://files.lahfa.xyz/nixos/nixos-minimal-23.11pre130979.gfedcba-x86_64-linux.iso

**Comment by hacker1024** [comment]
I have noticed the same issue.

Crashes after loading GRUB UEFI executable:
- Dell Inspiron 7586
- Gigabyte GA-B250M-HD3
- Microsoft Surface Pro 3

Boots successfully:
- Apple Macbook8,1
- Oracle VM VirtualBox

**Comment by RaitoBezarius** [comment]
Can you retry on the crashed platform with the ISO I provide above?

**Comment by RaitoBezarius** [comment]
> @RaitoBezarius with systemd-boot? I use systemd-boot from nixos-unstable on my laptop (the Inspiron 7586) already. I have noticed no issues.

Yes, but this is an install media now :). It would help to know if this works out of the box.

**Comment by chasecaleb** [comment]
@RaitoBezarius Nice job! That fixes the issue for me. I can't boot the latest nixos-unstable minimal ISO ( https://releases.nixos.org/nixos/unstable/nixos-23.11pre509514.9e1960bc196b/nixos-minimal-23.11pre509514.9e1960bc196b-x86_64-linux.iso ) on VMware Workstation with UEFI, which has been an ongoing issue for roughly a month or so. The ISO you provided with systemd works :tada:

**Comment by evalgrind** [comment]
Also for those who don't want to build the ISO, as a workaround you can also boot into an older version of the ISO (e.g. 23.05) where the specific GRUB commit doesn't exist.

**Comment by giantdwarf17** [comment]
Just encountered the same issue. Installing via pikvm and got to this screen: 

![Image](https://github.com/user-attachments/assets/a132da51-329c-4f4b-8404-cd04468247bc)

But after selecting install it just dropped me straight into a black screen. Tried all of the options in the menu to the same effect.

Reverting to 23.05 worked.

**Comment by K900** [comment]
Why are you using 23.11?
